awstesting/integration/smoke/shared.go

```diff
@@ -126,6 +131,87 @@ func init() {
 		assert.False(T, ok, "error returned")
 		assert.NoError(T, err)
 	})
+
+	When(`^I get all fixtures for "(.+?)" "(.+?)" from "(.+?)"$`,
+		func(t, cekAlg, bucket string) {
+			prefix := "plaintext_test_case_"
+			baseFolder := "crypto_tests/" + cekAlg
+			s3Client := World["client"].(*s3.S3)
+
+			out, err := s3Client.ListObjects(&s3.ListObjectsInput{
+				Bucket: aws.String(bucket),
+				Prefix: aws.String(baseFolder + "/" + prefix),
+			})
+			assert.NoError(T, err)
+
+			cases := []string{}
+			plaintexts := [][]byte{}
+			for _, obj := range out.Contents {
+				plaintextKey := obj.Key
+				ptObj, err := s3Client.GetObject(&s3.GetObjectInput{
+					Bucket: aws.String(bucket),
+					Key:    plaintextKey,
+				})
+				assert.NoError(T, err)
+				caseKey := strings.TrimPrefix(*plaintextKey, baseFolder+"/"+prefix)
+				cases = append(cases, caseKey)
+				plaintext, err := ioutil.ReadAll(ptObj.Body)
+				assert.NoError(T, err)
+				plaintexts = append(plaintexts, plaintext)
+			}
+			World["cases"] = cases
+			World["baseFolder"] = baseFolder
+			World["bucket"] = bucket
+			World["plaintexts"] = plaintexts
+		})
+
+	Then(`^I decrypt each fixture against "(.+?)" "(.+?)"$`, func(language, version string) {
+		cases := World["cases"].([]string)
+		baseFolder := World["baseFolder"].(string)
+		bucket := World["bucket"].(string)
+		prefix := "ciphertext_test_case_"
+		s3Client := World["client"].(*s3.S3)
+		s3CryptoClient := World["cryptoClient"].(*s3crypto.Client)
+		language = "language_" + language
+
+		ciphertexts := [][]byte{}
+		for _, caseKey := range cases {
+			cipherKey := baseFolder + "/" + version + "/" + language + "/" + prefix + caseKey
+
+			// To get metadata for encryption key
+			ctObj, err := s3Client.GetObject(&s3.GetObjectInput{
+				Bucket: aws.String(bucket),
+				Key:    &cipherKey,
+			})
+			assert.NoError(T, err)
+
+			masterkeyB64 := ctObj.Metadata["Masterkey"]
+			masterkey, err := base64.StdEncoding.DecodeString(*masterkeyB64)
+			assert.NoError(T, err)
+			cipher, err := s3crypto.NewAESECB(masterkey)
+			s3CryptoClient.MasterKey = cipher
+			ctObj, err = s3CryptoClient.GetObject(s3crypto.NewAESCBC, &s3crypto.GetObjectInput{
```

Comment:
I think we can compress the get metadata into the crypto client's GetObject class. Since the request's response will include the headers as well, can just read them prior to deprecating the content.